File: planner/app/src/main/java/com/pro3/planner/LocalSettingsManager.java
package com.pro3.planner;

import android.content.SharedPreferences;

/**
 * Created by linus_000 on 18.11.2016.
 */

public class LocalSettingsManager {

    private String sortingMethod;
    private static LocalSettingsManager instance;
    private SharedPreferences prefs;

    public static LocalSettingsManager getInstance() {
        if (instance == null) {
            instance = new LocalSettingsManager();
        }
        return instance;
    }

    public void setPrefs(SharedPreferences prefs) {
        this.prefs = prefs;
    }

    public void addCategory(String categoryName) {
        int categoryVisibility = prefs.getInt(categoryName, 0);

        SharedPreferences.Editor editor = prefs.edit();
        editor.putInt(categoryName, categoryVisibility);
        editor.commit();
    }

    public void setCategoryVisibility(String categoryID, int visibility) {
        SharedPreferences.Editor editor = prefs.edit();
        editor.putInt(categoryID, visibility);
        editor.commit();
    }

    public int getCategoryVisibility(String categoryID) {
        return prefs.getInt(categoryID, 1);
    }

    public void removeCategory(String categoryID) {
        SharedPreferences.Editor editor = prefs.edit();
        editor.remove(categoryID);
        editor.apply();
    }

    public void setColorVisibility(int color, int visibility) {
        SharedPreferences.Editor editor = prefs.edit();
        editor.putInt(Integer.toString(color), visibility);
        editor.commit();
    }


    public int getColorVisibility(int color) {
        return prefs.getInt(Integer.toString(color), 1);
    }

    public void setSortingMethod(String sortingMethod) {
        SharedPreferences.Editor editor =  prefs.edit();
        editor.putString("mainElementSorting", sortingMethod);
        this.sortingMethod = sortingMethod;
        editor.commit();
    }

    public String getSortingMethod() {
        if (this.sortingMethod != null) {
            return this.sortingMethod;
        } else {
            return prefs.getString("mainElementSorting", "nameAscending");
        }
    }
}

File: planner/app/src/main/java/com/sunilson/firenote/presentation/shared/singletons/LocalSettingsManager.kt
package com.sunilson.firenote.presentation.shared.singletons

import android.app.Application
import android.content.Context
import android.content.SharedPreferences
import org.apache.commons.codec.binary.Hex
import org.apache.commons.codec.digest.DigestUtils
import java.io.UnsupportedEncodingException
import java.security.NoSuchAlgorithmException
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class LocalSettingsManager @Inject constructor(context: Context) {

    private val sharedPrefs: SharedPreferences = context.getSharedPreferences("prefs", Context.MODE_PRIVATE)

    fun setCategoryVisiblity(id: String, visibility: Int) {
        val editor = sharedPrefs.edit()
        editor.putInt(id, visibility)
        editor.commit()
    }

    fun setColorVisibility(color: Int, visibility: Int) {
        val editor = sharedPrefs.edit()
        editor.putInt(color.toString(), visibility)
        editor.commit()
    }

    fun setSortingMethod(sortingMethod: String) {
        val editor = sharedPrefs.edit()
        editor.putString("mainElementSorting", sortingMethod)
        editor.commit()
    }

    fun setMasterPassword(password: String) {
        val editor = sharedPrefs.edit()
        editor.putString("masterPassword", password)
        editor.commit()
    }

    fun getMasterPassword() = sharedPrefs.getString("masterPassword", "");
    fun getSortingMethod() = sharedPrefs.getString("mainElementSorting", null)
    fun getColorVisibility(color: Int) = sharedPrefs.getInt(color.toString(), -1)
    fun getCategoryVisibility(category: String) = sharedPrefs.getInt(category, -1)
    fun getSHA1Hash(masterPasswordHash: String): String = String(Hex.encodeHex(DigestUtils.sha1(masterPasswordHash)))

}